src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngine.cs

```diff
@@ -240,7 +240,13 @@ private ViewLocationCacheResult LocatePageFromPath(string executingFilePath, str
         {
             var controllerName = GetNormalizedRouteValue(actionContext, ControllerKey);
             var areaName = GetNormalizedRouteValue(actionContext, AreaKey);
-            var razorPageName = GetNormalizedRouteValue(actionContext, PageKey);
+            string razorPageName = null;
+            if (actionContext.ActionDescriptor.RouteValues.ContainsKey(PageKey))
+            {
+                // Only calculate the Razor Page name if "page" is registered in RouteValues.
+                razorPageName = GetNormalizedRouteValue(actionContext, PageKey);
```

Comment:
Ideally I'd like to change how `GetNormalizedRouteValue` works to verify it returns a value only if it's present in `RouteValues` but that would be too wide a change for 2.0.1. We could consider it for 2.1.0 though.